message_ix/core.py

```diff
@@ -58,6 +58,37 @@ def _init_scenario(s, commit=False):
                 }),
             ],
         },
+        {  # required for addon formulation
+            'test': 'addon' not in s.set_list(),
+            'exec': [
+                (s.init_set, {'args': ('addon',)}),
+                (s.init_set, {'args': ('type_addon',)}),
+                (s.init_set, {'args': ('cat_addon', ['type_addon', 'addon'])}),
+                (s.init_set, {
+                    'args': ('map_tec_addon', ['technology', 'type_addon'])
+                }),
+                (s.init_par, {
+                    'args': (
+                        'addon_conversion',
+                        ['node', 'technology', 'addon',
```

Comment:
this should be indexed on `type_addon`, not `addon` - please double-check with @OFR-IIASA to make sure this is the latest version of the code that actually worked for the many-to-many addon mapping.